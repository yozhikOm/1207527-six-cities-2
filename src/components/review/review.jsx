import React from 'react';
import PropTypes from 'prop-types';

const Review = (props) => {
  const {review} = props;

  return (
    <li className="reviews__item">
      <div className="reviews__user user">
        <div className="reviews__avatar-wrapper user__avatar-wrapper">
          <img className="reviews__avatar user__avatar" src={review.user.avatarUrl} width="54" height="54" alt="Reviews avatar" />
        </div>
        <span className="reviews__user-name">
          {review.user.name}
        </span>
      </div>
      <div className="reviews__info">
        <div className="reviews__rating rating">
          <div className="reviews__stars rating__stars">
            <span style={{width: `${review.rating}%`}}></span>
            <span className="visually-hidden">Rating</span>
          </div>
        </div>
        <p className="reviews__text">
          {review.comment}
        </p>
        <time className="reviews__time" dateTime="2019-04-24">{review.date}</time>
      </div>
    </li>

  );
};

Review.propTypes = {
  review: PropTypes.shape({
    id: PropTypes.number.isRequired,
    user: PropTypes.shape({
      id: PropTypes.number,
      isPro: PropTypes.bool,
      name: PropTypes.string,
      avatarUrl: PropTypes.string,
    }),
    rating: PropTypes.number.isRequired,
    comment: PropTypes.string,
    date: PropTypes.date,

  }).isRequired
};

export {Review};
