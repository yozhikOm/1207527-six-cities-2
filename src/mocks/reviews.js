export const reviews = [
  {
    id: 1,
    user: {
      id: 1,
      isPro: true,
      name: `Erika`,
      avatarUrl: ``,
    },
    rating: 4.6,
    comment: `review comment`,
    date: `Июль 2018`,
  },

];
