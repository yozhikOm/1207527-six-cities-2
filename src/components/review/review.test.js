import React from 'react';
import {Review} from './review.jsx';
import renderer from 'react-test-renderer';
import {reviews} from '../../mocks/reviews.js';

it(`Review компонент рендерится корректно`, () => {
  const reviewComponent = renderer.create(<Review review={reviews[0]}/>).toJSON();
  expect(reviewComponent).toMatchSnapshot();
});
