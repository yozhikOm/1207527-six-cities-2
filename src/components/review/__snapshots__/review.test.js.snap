// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review компонент рендерится корректно 1`] = `
<li
  className="reviews__item"
>
  <div
    className="reviews__user user"
  >
    <div
      className="reviews__avatar-wrapper user__avatar-wrapper"
    >
      <img
        alt="Reviews avatar"
        className="reviews__avatar user__avatar"
        height="54"
        src=""
        width="54"
      />
    </div>
    <span
      className="reviews__user-name"
    >
      Erika
    </span>
  </div>
  <div
    className="reviews__info"
  >
    <div
      className="reviews__rating rating"
    >
      <div
        className="reviews__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "4.6%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <p
      className="reviews__text"
    >
      review comment
    </p>
    <time
      className="reviews__time"
      dateTime="2019-04-24"
    >
      Июль 2018
    </time>
  </div>
</li>
`;
